adoptingerlang.org

```diff
@@ -4101,20 +4101,118 @@ With these broad lines covered, we can start exploring the processes and practic
+- How do you approach structuring, writing, and debugging code in the language? Are there basic guidelines or tools that should be provided?
+- How do you hire for these developers? How do you train those who do not know about the language?
+- Do we have any basic on-boarding documentation to help people get started with any of the factors above?
+
+Add or remove questions as you see fit. You can contextualize some like memory footprint, requirements for front-end languages, mobile development, back-end languages, desktop software, and so on. Come up with a baseline your engineers think should be provided. Do it in the abstract, without thinking of a single programming language. Classify some of these things as critical, nice to have, and so on. You might come up with a table looking like this:
+
+| Item | Critical | Nice to Have | In an Ideal World |
+|------+---------+----------+--------------+-------|
+| HTTP Server | bare server | web framework | popular web framework with community support |
+| gRPC Library  | serialization/deserialization | full gRPC client/server | well-documented examples |
+| Micro services | sample implementation | templates library | full tutorial |
+| Number of local Experts | 2 | 4 | 10+ |
+| Hireability | sizeable local user group | taught in college | everyone knows it |
+| On-boarding | people pick things up on their own | some documentation exists | tutorials and training provided |
+| Observability | logging and metrics library | guidelines on language- or framework-specific logs/metrics | open telemetry support |
+| Testing | unit testing framework | integration testing framework | advanced testing frameworks (Property-Based Testing, mutation testing, etc.) |
+| ... | ... | ... | ... |
+
+Establish a base criteria to hit, the critical items without which you can't adopt the language, and so on. Turn it into a rather objective scorecard you can use to validate decisions. Otherwise, what will happen is that a CTO or architect will push for whatever tech they fancy, without regards to actual support. The default reflex of organizations is actually to just find something someone higher up likes, and push for it and let the rest of the organization adjust to these whims. A score card (as simple or complex as you want it to be) aims to make people aware of the costs of adopting or changing technologies, and as a checklist of what they need to provide to help each other use it.
+
+Run the scorecard against a language you want to adopt, and see if you could indeed adopt it successfully or what might be missing to do it well; run the scorecard against languages you already have adopted, and see if they would pass. If they don't, start by giving the support you do not have right now. Grow a knowledge base that is guided by what your engineers feel they'd need to do their job more effectively. Specialize and adapt the requirements for some areas of the company, and see where this lands you.
```

Comment:
```suggestion
Run the scorecard against a language you want to adopt, and see if you could indeed adopt it successfully or what might be missing to do it well; run the scorecard against languages you already have adopted, and see if they would pass. If they don't, start by fixing the gaps. Grow a knowledge base that is guided by what your engineers feel they'd need to do their job more effectively. Specialize and adapt the requirements for some areas of the company, and see where this lands you.
```

I wasn't sure how to fix that sentence so just changed it to "fix the gaps" -- wasn't sure if you were trying to say 'giving the support' like give man power to this stuff or simply saying to make the language meet all the criteria.